### ISSUE TITLE
Production: How to share Inferentia cores between different models?

### ISSUE THREAD

**Issue by Matthieu-Tinycoaching** [aws/aws-neuron-sdk#289]
Hi,

I was wondering how to share inferentia cores between different pytorch models in production? Do we have to put all models within a container and specify environment variables adequately for each model?

How to allocate some of the cores to a specific models and the others for others models?

Thanks!

**Comment by jeffhataws** [comment]
Hi Matthieu,

Please take a look at the [guide for configuring NeuronCore groups](https://awsdocs-neuron.readthedocs-hosted.com/en/latest/neuron-guide/neuron-frameworks/tensorflow-neuron/tutorials/tutorial-tensorflow-NeuronCore-Group.html). While it is written for TensorFlow-Neuron, the concepts described there applies to PyTorch-Neuron also. If it doesn't answer your questions, we will be happy to work closely with you to unblock your progress.

**Comment by Matthieu-Tinycoaching** [comment]
Hi @jeffhataws thanks for your feedback.

The link is useful, however how to apply these guidelines with model server as torchserve or bentoML which I use?

**Comment by nonatofabio** [comment]
Hi Matthieu,

Please check out the [BERT TorchServe Tutorial in the Neuron Documentation](https://awsdocs-neuron.readthedocs-hosted.com/en/latest/neuron-guide/neuron-frameworks/pytorch-neuron/tutorials/tutorial-torchserve.html). It shows the integration of Neuron SDK and TorchServe, including defining `NEURONCORE_GROUP_SIZES` in the model handler script and use it while scaling TorchServe workers. Please let us know if this addresses your questions regarding configuring NeuronCore Groups for TorchServe.

**Comment by aws-joshim** [comment]
Hi Matthieu,
Closing this ticket since the BERT TorchServe tutorial and Neuron Core groups configuration was shared in the thread above. Please create a new ticket or reopen if you have further questions
